src/Runtime/SymbolResolver.cs

```diff
@@ -55,11 +55,36 @@ static SymbolResolver ()
             }
         }
 
-        public static IntPtr LoadImage (ref string name)
+        public static IntPtr LoadImage (ref string name, string searchdir = null)
         {
+            if (searchdir == null)
+                searchdir = Environment.CurrentDirectory;
+
             foreach (var format in formats)
             {
-                var attempted = System.IO.Path.Combine(Environment.CurrentDirectory, string.Format (format, name));
+                // Search the Current or specified directory for the library
+                string filename = string.Format(format, name);
+                string attempted = System.IO.Path.Combine(searchdir, filename);
+                if (System.IO.File.Exists(attempted) == false)
+                {
+                    // Search the Path directories for the library
+                    var pathvalues = Environment.GetEnvironmentVariable("PATH");
```

Comment:
Please move this outside the loop since it's always the same - this way there are many redundant calls.